src/assets/Standards/CollectibleStandards/ERC721/ERC721Standard.ts

```diff
@@ -221,46 +221,45 @@ export class ERC721Standard {
     name: string | undefined;
     image: string | undefined;
   }> => {
-    const [isERC721, supportsMetadata] = await Promise.all([
-      this.contractSupportsBase721Interface(address),
-      this.contractSupportsMetadataInterface(address),
-    ]);
-    let tokenURI, symbol, name, image;
-    if (supportsMetadata) {
+    const isERC721 = await this.contractSupportsBase721Interface(address);
+    if (!isERC721) {
+      throw new Error("This isn't a valid ERC721 contract");
+    }
+
+    let tokenURI, image, symbol, name;
+    try {
       [symbol, name] = await Promise.all([
```

Comment:
If either of these promises fail, we won't get the other one, is that intentional?

Reply:
No intentional. Good catch! Will fix

Reply:
fixed here: https://github.com/MetaMask/controllers/pull/834/commits/3e0d99fb06dc5b5e2eb79a8f7d418cb8747a5a45

Reply:
This will have the same problem though, right?

Reply:
Or at least if the symbol fails you never get the name. Might be fine though, not sure 🤔

Reply:
smh yes you're right... I'm being a ditz today.

Reply:
re-fixed here: https://github.com/MetaMask/controllers/pull/834/commits/dc274b128edc23437ba4c7dbecc2cfed92718362

Reply:
👍